### ISSUE TITLE
Restrict devices for a provisioning profile through Match

### ISSUE THREAD

**Issue by ribtiago** [fastlane/fastlane#11660]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm using Match to generate provisioning profiles and I would like to have the possibility of specifying the list of devices when I generate a certain profile.
This is possible through the Apple Developer website, but I can't find a way to do it through Fastlane.

My suggestion would be something like a file that Match could use to register those devices if they are not yet registered, and associate only the devices on that file to the provisioning profile.

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.2                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.2 (17C88)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby

default_platform :iOS

platform :ios do

	before_all do | lane |
		ensure_git_status_clean
	end

	after_all do | lane |
		if lane != :start_hotfix
			push_to_git_remote
		end
	end

	error do | lane, exception |
		reset_git_repo(skip_clean: true)
	end

	desc "Creates a beta build on HockeyApp (app: Hidden Beta)"
	desc ""
	desc "If 'new_version' is provided, a version bump will be done"
	desc ""
	desc "Options:"
	desc "*new_version* Optional. If provided, can be 'major' or 'minor'"
	desc ""
	desc "Usage 'fastlane beta (new_version:[major|minor])'"
	lane :beta do | options |

		ensure_git_branch(branch: "develop")


		if options[:new_version]
			if ["major", "minor"].include? options[:new_version]
				increment_version_number(bump_type: options[:new_version])
			else
				UI.user_error! "A valid bump type was not provided. Acceptable options for new_version are 'major' or 'minor'"
			end
		end

		increment_build_number

		disable_ipad

		match(type: "adhoc")
		gym(scheme: "Hidden", export_method: "ad-hoc")
		hockey(api_token: ENV["HOCKEY_BETA_API_TOKEN"], notify: "0")

		enable_ipad

		commit_version_bump(xcodeproj: "Hidden.xcodeproj")
	end

	desc "Creates a public release build on HockeyApp (app: Hidden)"
	desc ""
	desc "Usage 'fastlane release'"
	lane :release do

		git_flow_release_start

		disable_ipad

		match(type: "adhoc")
		gym(scheme: "Hidden", export_method: "ad-hoc")
		hockey(api_token: ENV["HOCKEY_RELEASE_API_TOKEN"], notify: "0")

		enable_ipad

		git_flow_release_finish
	end

	desc "Starts a new branch to develop an hotfix, based on the current master status"
	desc ""
	desc "Usage 'fastlane start_hotfix'"
	lane :start_hotfix do

		git_flow_hotfix_start

		increment_build_number
		increment_version_number(bump_type: "patch")
		commit_version_bump(xcodeproj: "Hidden.xcodeproj")

		UI.message "You can now start developing the hotfix"
		UI.message "When finished use 'fastlane apply_hotfix' to generate a new hotfix version"
	end

	desc "Applies the current hotfix in development and releases to HockeyApp"
	desc "In the end merges the hotfix branch into master and develop and deletes it"
	desc ""
	desc "Options:"
	desc "*skip_internal*			Optional boolean"
	desc "*skip_public*				Optional boolean"
	desc ""
	desc "Usage 'fastlane apply_hotfix (skip_internal:[true:false]) (skip_public:[true:false])'"
	lane :apply_hotfix do | options |

		skip_internal = options.fetch(:skip_internal, false)
		skip_public = options.fetch(:skip_public, false)

		ensure_git_branch(branch: "hotfix/#{get_version_number}")

		disable_ipad

		match(type: "adhoc")
		gym(scheme: "Hidden", export_method: "ad-hoc")

		if !skip_internal
			UI.message "Releasing to private HockeyApp"
			hockey(api_token: ENV["HOCKEY_BETA_API_TOKEN"], notify: "0")
		end

		if !skip_public
			UI.message "Releasing to public HockeyApp"
			hockey(api_token: ENV["HOCKEY_RELEASE_API_TOKEN"], notify: "0")
		end

		enable_ipad

		git_flow_hotfix_finish
	end

	# enable/disables ipad support
	def disable_ipad
		sh "sed -i '' 's/TARGETED_DEVICE_FAMILY = \"1,2\";/TARGETED_DEVICE_FAMILY = 1;/g' ../Hidden.xcodeproj/project.pbxproj"
	end

	def enable_ipad
		sh "sed -i '' 's/TARGETED_DEVICE_FAMILY = 1;/TARGETED_DEVICE_FAMILY = \"1,2\";/g' ../Hidden.xcodeproj/project.pbxproj"
	end

	# git flow helpers
	def git_flow_release_start
		ensure_git_branch(branch: "develop")
		sh("git", "flow", "release", "start", get_version_number)
	end

	def git_flow_release_finish
		sh("git", "flow", "release", "finish", "-m", "\"Released #{get_version_number}\"")
		push_git_tags
	end

	def git_flow_hotfix_start

		# we don't want to increment the version before starting the hotfix
		# but there is no other way to get the next version programatically,
		# so we are incrementing the version and then reseting the repo to the current clean state
		new_version = increment_version_number(bump_type: "patch")
		reset_git_repo(skip_clean: true)

		sh("git", "flow", "hotfix", "start", new_version)
	end

	def git_flow_hotfix_finish
		sh("git", "flow", "hotfix", "finish", "-m", "\"Released hotfix #{get_version_number}\"")
		push_git_tags
	end
end

## Utilities

# setup lane

desc "Setups a new development machine"
desc ""
desc ""
desc "Usage 'fastlane setup'"
lane :setup do
	match(type: "development")
	carthage(platform: "iOS")
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.company.Hidden" 
apple_id "my.email@company.com"

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.78.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.3        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.0.15       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| libxml-ruby               | 3.0.0        |
</details>

*generated on:* **2018-01-23**
</details>

**Comment by joshdholtz** [comment]
@ribtiago Thanks for the feature suggestion! This is something we will think about adding when we get time unless you are willing to attempt to make this change on your own and open up a PR for it 😊
